商品詳細表示

K-Hir-pk18 opened the pull request:
WHAT
・商品詳細機能を実装

WHY
・出品者は商品の編集のために必要
・購入希望者は判断んおために必要

K-Hir-pk18 committed:
商品詳細表示

WHAT
・商品詳細機能を実装

WHY
・出品者は商品の編集のために必要
・購入希望者は判断んおために必要

K-Hir-pk18 commented:
ログインユーザー
https://gyazo.com/4035e56f599899c2ba96f619788e26b9

非ログインユーザー
https://gyazo.com/757dd7485ddfbe9e2b1b988a937c2231

K-Hir-pk18 commented:
ご担当者様


よろしくお願いいたします。

masayaomoto commented:
レビュー依頼ありがとうございます！

実際の画面や機能ができているか確認させていただきたいので、
追加で下記キャプチャ動画をご提出いただけますか？
```
・ログイン状態の出品者以外のユーザーのみ、「購入画面に進むボタン」が表示される動画
```
よろしくお願いいたします。

K-Hir-pk18 commented:
masayaomoto様

レビューありがとうございます。
先のものとあわせ、以下の通りとなります。
ご確認よろしくお願い致します。

出品者（編集/削除）
https://gyazo.com/4035e56f599899c2ba96f619788e26b9

非ログインユーザー(ボタンなし)
https://gyazo.com/757dd7485ddfbe9e2b1b988a937c2231

他ログインユーザー(購入ボタン)
https://gyazo.com/62cb76e8cae8b8a0765d410de00c4505

K-Hir-pk18 commented:
ご担当者様

よろしくお願いいたします。

chisesakai commented on app/controllers/products_controller.rb:
```diff
@@ -11,6 +11,11 @@ def new
   end
 
+  def show
+    @product = Product.new
```
```ruby
    @product = Product.new
```
新しく生成する必要はないため、こちらのコードは削除しましょう。

chisesakai commented on app/views/products/show.html.erb:
```diff
@@ -4,66 +4,65 @@
     </h2>
     <div class="item-img-content">
-      <%= image_tag "item-sample.png" ,class:"item-box-img" %>
+      <%= image_tag @product.image, class: "item-box-img" if @product.image.attached? %>
```
```ruby
      <%= image_tag @product.image, class: "item-box-img" if @product.image.attached? %>
```
`@product.image.attached?`こちらは削除しましょう！
画像は常に存在するためです。

chisesakai commented on app/views/products/show.html.erb:
```diff
@@ -4,66 +4,65 @@
     <%= link_to "商品の編集", "#", method: :get, class: "item-red-btn" %>
     <p class="or-text">or</p>
     <%= link_to "削除", "#", method: :delete, class:"item-destroy" %>
+  <% end %>
 
-    <%# 商品が売れていない場合はこちらを表示しましょう %>
-    <%= link_to "購入画面に進む", "#" ,class:"item-red-btn"%>
-    <%# //商品が売れていない場合はこちらを表示しましょう %>
-
+  <% if user_signed_in? && current_user.id != @product.user_id %> <%# ＆売買済%>
+      <%= link_to "購入画面に進む", "#" ,class:"item-red-btn"%>
+  <% end %>
```
```ruby
  <% if user_signed_in? && current_user.id == @product.user_id %> <%# &売買済%>
    <%= link_to "商品の編集", "#", method: :get, class: "item-red-btn" %>
    <p class="or-text">or</p>
    <%= link_to "削除", "#", method: :delete, class:"item-destroy" %>
  <% end %>

  <% if user_signed_in? && current_user.id != @product.user_id %> <%# ＆売買済%>
      <%= link_to "購入画面に進む", "#" ,class:"item-red-btn"%>
  <% end %>
```
if文をまとめてみましょう！


- ` user_signed_in? `は全てにかかっている
- 編集削除は出品者、購入は出品者でない
ですので、下記のような構成になると良いですね。
購入機能実装後の分岐も追加しやすくなるかと思います。
```
if user_signed_in? 
  if出品者
    編集削除
  else
    購入
  end
end 
```

chisesakai commented on app/views/products/show.html.erb:
```diff
@@ -103,7 +102,7 @@
     </a>
   </div>
   <%# 詳細ページで表示されている商品のカテゴリー名を表示しましょう %>
```
```
  <%# 詳細ページで表示されている商品のカテゴリー名を表示しましょう %>
```
```
<%# //詳細ページで表示されている商品のカテゴリー名を表示しましょう %>
```
こちらは実装のためのご案内ですので削除お願い致します。

chisesakai commented on config/routes.rb:
```diff
@@ -2,6 +2,6 @@
   root to: "products#index"
   resources :users
-  resources :products, only: [:index, :new, :post, :create]
+  resources :products, only: [:index, :new, :post, :create, :show]
```
```ruby
  resources :products, only: [:index, :new, :post, :create, :show]
```
` :post`こちらはそのような名前のアクションの定義もなく、不要かと思いますので削除しましょう！

K-Hir-pk18 replied:
chisesakai様

レビューありがとうございます。
こちら対応して再度レビューをお願いいたします。

chisesakai reviewed (commented):
レビュー依頼ありがとうございます。学習お疲れさまです！
キャプチャ拝見させていただきました。改めてのご提出、ありがとうございます！
#### 5件コメントいたしましたので、ご確認お願いいたします。

引き続き一緒に頑張っていきましょう！
修正頂いた後、コミットプッシュを確認いただき、再度レビュー依頼のご提出をお願い致します！

K-Hir-pk18 committed:
商品詳細表示レビュー対応

WHAT
レビュー対応
・@product = Product.new　の削除
・app/views/products/show.html.erb
@product.image.attached?の削除
・app/views/products/show.html.erbのif文
・app/views/products/show.html.erbコメントアウト
・config/routes.rb修正

WHY
より良いコードとするため

K-Hir-pk18 commented:
ご担当者様

よろしくお願いいたします。

K-Hir-pk18 commented:
出品者
https://gyazo.com/067d682897bd3464b8d9dea7804326b3

他ユーザー
https://gyazo.com/59d2fd22b91114db49c4531d7abbbb2e

非ログインユーザー
https://gyazo.com/6f5db02095126b42582279723c8d58d9

ky3090 reviewed (commented):
レビュー依頼並びに修正ありがとうございます。
LGTMとさせていただきます。実装お疲れさまでした。

マージ後、ローカルのmasterブランチに忘れずにプルしましょう。
その後、新しいブランチを作成し次の実装に取り組んでいただけますと幸いです。
引き続き頑張っていきましょう。

K-Hir-pk18 commented:
ky3090様

ギリギリの時間までご対応いただき本当に感謝です！！
ありがとうございます。

これで枕高くして寝れます。
…もうちょっとガンバります！
